sqldb/sqlc/queries/invoices.sql

```diff
@@ -118,25 +143,14 @@ WHERE invoice_id = $1;
 
 -- name: UpdateInvoiceHTLC :exec
 UPDATE invoice_htlcs 
-SET state=$2, resolve_time=$3
-WHERE id = $1;
+SET state=$4, resolve_time=$5
+WHERE htlc_id = $1 AND chan_id = $2 AND invoice_id = $3;
 
 -- name: UpdateInvoiceHTLCs :exec
 UPDATE invoice_htlcs 
 SET state=$2, resolve_time=$3
 WHERE invoice_id = $1 AND resolve_time IS NULL;
 
-
--- name: DeleteInvoiceHTLC :exec
-DELETE
-FROM invoice_htlcs
```

Comment:
If we cancel back AMP payment due to failure, then we may still need to delete the HTLCs.

Reply:
Ah yeh today we just end up marking as cancelled on disk, see  `cancelHtlcsAmp`. Then eventually the invoice GC loop should tick if active to delete them.